pkg/build/dir_structure.go

```diff
@@ -0,0 +1,63 @@
+package build
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+const (
+	// scriptsSubDir contains user provided custom scripts.
+	scriptsSubDir = "scripts"
+	// rpmsSubDir contains user provided RPM files.
+	rpmsSubDir = "rpms"
+)
+
+type DirStructure struct {
+	// ImageConfigDir is the root directory storing all configuration files.
+	ImageConfigDir string
+	// BuildDir is the directory used for assembling the different components used in a build.
+	BuildDir string
+	// CombustionDir is a subdirectory under BuildDir containing the Combustion script and all related files.
+	CombustionDir string
+	// ScriptsDir is an optional subdirectory under ImageConfigDir containing user provided custom scripts.
+	ScriptsDir string
+	// RPMDir is an optional subdirectory under ImageConfigDir containing user provided RPM files.
+	RPMDir string
+	// DeleteBuildDir indicates whether the BuildDir should be cleaned up after the image is built.
+	DeleteBuildDir bool
+}
+
+func NewDirStructure(imageConfigDir, buildDir string, deleteBuildDir bool) (*DirStructure, error) {
+	if buildDir == "" {
+		tmpDir, err := os.MkdirTemp("", "eib-")
+		if err != nil {
+			return nil, fmt.Errorf("creating a temporary build directory: %w", err)
+		}
+		buildDir = tmpDir
+	}
+	combustionDir := filepath.Join(buildDir, "combustion")
+
+	if err := os.MkdirAll(combustionDir, os.ModePerm); err != nil {
+		return nil, fmt.Errorf("creating the combustion directory: %w", err)
+	}
+
+	return &DirStructure{
+		ImageConfigDir: imageConfigDir,
+		BuildDir:       buildDir,
+		CombustionDir:  combustionDir,
+		ScriptsDir:     filepath.Join(imageConfigDir, scriptsSubDir), // may not exist
+		RPMDir:         filepath.Join(imageConfigDir, rpmsSubDir),    // may not exist
```

Comment:
Added these two in order to avoid separate functions (and potentially future components) having to assemble them separately. Perhaps we could also add a check whether they exist instead of doing it in the `configureXX()` functions. Thoughts?

Reply:
These will be dropped and the type will be renamed to `Context` which the different components will use.